4-WebApp-your-API/4-1-MyOrg/README-use-keyvault-certificate.md

```diff
@@ -0,0 +1,77 @@
+```json
+"ClientCertificates": [
+  {
+    "SourceType": "KeyVault",
+    "KeyVaultUrl": "[Enter URL for you KeyVault]",
+    "KeyVaultCertificateName": "[Enter name of the certificate]"
+  }
+]
+```
+4. Set exection policy
+```powershell
+Set-ExecutionPolicy -ExecutionPolicy RemoteSigned -Scope Process -Force
+```
+5. Run the cleanup script to delete old App Registration for the sample
+```powershell
+AppCreationScripts-withCert/Cleanup.ps1
+```
+6. Run the configuration script to re-create the App Registration. The script will also create a [application name].pfx file that will be **manually** uploaded into KeyVault. When asked about a password, remember it - you will need the password when uploading the certificate.
+```powershell
+AppCreationScripts-withCert/Configuration.ps1
+```
+7. If you still don't have KeyVault, sign in to the [Azure portal](https://portal.azure.com) and [create an Azure KeyVault](https://docs.microsoft.com/en-us/azure/key-vault/general/quick-create-portal)
+8. Inside Client/appsettings.json file - update "KeyVaultUrl" key to have URL of your KeyVault
+9. [Upload](https://docs.microsoft.com/en-us/azure/key-vault/certificates/tutorial-import-certificate#import-a-certificate-to-key-vault) the generated AppCreationScripts-withCert\.PFX file into the KeyVault
+10. Run the sample as indicated in [README.md](README.md)
+
+## More information
+
+### About Azure Key Vault
+
+Cloud applications and services use cryptographic keys and secrets to help keep information secure. [Azure Key Vault](https://azure.microsoft.com/services/key-vault/) safeguards these keys and secrets. When you use Key Vault, you can encrypt authentication keys, storage account keys, data encryption keys, .pfx files, and passwords by using keys that are protected by hardware security modules (HSMs).
+
+### About Managed Identities for Azure Resources
+
+Azure Key Vault provides a way to securely store credentials, secrets, and other keys, but your code has to authenticate to Key Vault to retrieve them. The [managed identities for Azure resources](https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/overview) feature in Azure Active Directory (Azure AD) solves this problem. The feature provides Azure services with an automatically managed identity in Azure AD. You can use the identity to authenticate to any service that supports Azure AD authentication, including Key Vault, without any credentials in your code.
+
+#### **Managed Identities for Azure Resources**
+
+In a daemon application scenario, Managed Identity will work if you have it deployed it in an [Azure Virtual Machine](https://azure.microsoft.com/services/virtual-machines/) or [Azure Web Job](https://docs.microsoft.com/en-us/azure/app-service/webjobs-create). Please, read [this documentation](https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/overview) to understand how Managed Identity works with an Azure VM.
```

Comment:
wecan remove this and place generic links about key vault and managed identity instead

Reply:
done